pilot/pkg/model/push_context.go

```diff
@@ -341,41 +343,6 @@ func (ps *PushContext) Services(proxy *Proxy) []*Service {
 	return out
 }
 
-// UpdateNodeIsolation will update per-node data holding visible services and configs for the node.
-// It is called:
-// - on connect
-// - on config change events (full push)
-// - TODO: on-demand events from Envoy
-func (ps *PushContext) UpdateNodeIsolation(proxy *Proxy) {
```

Comment:
Same - why are you removing this code ?

Reply:
The agreement was that we would remove all the CLI or flag driven scoping code when the API + implementation lands. And as it stands, this code is implementing a rudimentary version of sidecar scoping by auto scoping to individual namespaces and adding shared namespace through env flags, which conflicts with the sidecar implementation

Reply:
Sure - when the API lands this method needs to read the sidecar and add the user-defined egress.

We agreed to remove the flag to import a specific namespace - but we still need a flag to indicate that 'isolation' is enabled - because if a namespace has no Sidecar resource we need to know if we import all namespaces ( backward compat ) or none (new model).

Reply:
How about: leave the methods as they are, and I can make a separate PR to call the new Sidecar methods and add to the per-node list ?

Reply:
Each listener has a different set of Services per the sidecar API. These are not captured through this nodeIsolation thing. I will leave these as empty stubs with comments on what APIs to call if you want to play with the on-demand thing, but keep them as two mutually exclusive modes, both using the Sidecar API. Otherwise, I cannot guarantee any level of correctness inside the networking code when generating listeners or clusters, or selecting the right destination rule/virtualService, as there is a tricky relationship between the config scoping and the imports. (Its not enough to simply have list of services)

Reply:
That may work - we can start with getting on-demand as a mutually exclusive mode, and combine it later. 

And please add extensive comments on any 'tricky' relationship - the PR as is is extremely hard to understand and we need more people to be able to read the code and comments and be able to know what is going on. 

Maybe in 1.1 we can start with the simpler "default Sidecar only" - and focus on improving the 
code structure so it is understandable. We can live without the other selection, but I personally
don't understand all your code and its implications. Either add more comments until it is clear, 
or restructure it.

Reply:
To be fair, this is how I felt when you had the incremental EDS code and the EDS cache structure code. Just saying.. :)